packages/ts-client/src/webrtc/transciever.ts

```diff
@@ -0,0 +1,210 @@
+  serverTracks: Map<string, number>,
+) => {
+  const recvTransceivers = connection!
+    .getTransceivers()
+    .filter((elem) => elem.direction === 'recvonly');
+  let toAdd: string[] = [];
+
+  const getNeededTransceiversTypes = (type: string): string[] => {
+    let typeNumber = serverTracks.get(type);
+    typeNumber = typeNumber !== undefined ? typeNumber : 0;
+    const typeTransceiversNumber = recvTransceivers.filter(
+      (elem) => elem.receiver.track.kind === type,
+    ).length;
+    return Array(typeNumber - typeTransceiversNumber).fill(type);
+  };
+
+  const audio = getNeededTransceiversTypes('audio');
+  const video = getNeededTransceiversTypes('video');
+  toAdd = toAdd.concat(audio);
+  toAdd = toAdd.concat(video);
+
+  for (const kind of toAdd)
+    connection?.addTransceiver(kind, { direction: 'recvonly' });
+};
+
+export const setTransceiverDirection = (connection: RTCPeerConnection) => {
+  connection
+    .getTransceivers()
+    .forEach(
+      (transceiver) =>
+        (transceiver.direction =
+          transceiver.direction === 'sendrecv'
+            ? 'sendonly'
+            : transceiver.direction),
+    );
+};
+
+export const addTrackToConnection = <EndpointMetadata, TrackMetadata>(
+  trackContext: TrackContext<EndpointMetadata, TrackMetadata>,
+  disabledTrackEncodingsMap: Map<string, TrackEncoding[]>,
+  connection: RTCPeerConnection | undefined,
+) => {
+  const transceiverConfig = createTransceiverConfig(
+    trackContext,
+    disabledTrackEncodingsMap,
+  );
+  const track = trackContext.track!;
+  connection!.addTransceiver(track, transceiverConfig);
+};
+
+const createTransceiverConfig = <EndpointMetadata, TrackMetadata>(
+  trackContext: TrackContext<EndpointMetadata, TrackMetadata>,
+  disabledTrackEncodingsMap: Map<string, TrackEncoding[]>,
+): RTCRtpTransceiverInit => {
+  let transceiverConfig: RTCRtpTransceiverInit;
+
+  if (trackContext.track!.kind === 'audio') {
+    transceiverConfig = createAudioTransceiverConfig(trackContext);
+  } else {
+    transceiverConfig = createVideoTransceiverConfig(
+      trackContext,
+      disabledTrackEncodingsMap,
+    );
+  }
+
+  return transceiverConfig;
+};
+
+const createAudioTransceiverConfig = <EndpointMetadata, TrackMetadata>(
+  trackContext: TrackContext<EndpointMetadata, TrackMetadata>,
+): RTCRtpTransceiverInit => {
+  return {
+    direction: 'sendonly',
+    streams: trackContext.stream ? [trackContext.stream] : [],
+  };
+};
+
+const createVideoTransceiverConfig = <EndpointMetadata, TrackMetadata>(
+  trackContext: TrackContext<EndpointMetadata, TrackMetadata>,
+  disabledTrackEncodingsMap: Map<string, TrackEncoding[]>,
+): RTCRtpTransceiverInit => {
+  let transceiverConfig: RTCRtpTransceiverInit;
+  if (trackContext.simulcastConfig!.enabled) {
+    transceiverConfig = simulcastTransceiverConfig;
+    const trackActiveEncodings = trackContext.simulcastConfig!.activeEncodings;
+    const disabledTrackEncodings: TrackEncoding[] = [];
```

Comment:
this fn could be broken down to smaller functions to increase readibility

Reply:
I agree. I will refactor it in the future PRs